src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -213,12 +213,22 @@ public partial class HttpRequestMessage : System.IDisposable
         public System.Net.Http.Headers.HttpRequestHeaders Headers { get { throw null; } }
         public System.Net.Http.HttpMethod Method { get { throw null; } set { } }
         public System.Collections.Generic.IDictionary<string, object?> Properties { get { throw null; } }
+        public HttpRequestOptions Options { get { throw null; } }
         public System.Uri? RequestUri { get { throw null; } set { } }
         public System.Version Version { get { throw null; } set { } }
         public void Dispose() { }
         protected virtual void Dispose(bool disposing) { }
         public override string ToString() { throw null; }
     }
+
+    public readonly struct HttpRequestOptionsKey<TValue>
+    {
+        public HttpRequestOptionsKey(string key) {}
+        public string Key { get { throw null; } }
+    }
+
+    public sealed class HttpRequestOptions : System.Collections.Generic.Dictionary<string, object> { }
```

Comment:
In the proposal there are also these methods to get and set the typed values which are using the `HttpRequestOptionsKey` so you don't need to cast them [here](https://github.com/dotnet/runtime/pull/39182/files#diff-4353591640f3202f481b9b246f56a41fR132) and [here](https://github.com/dotnet/runtime/pull/39182/files#diff-4353591640f3202f481b9b246f56a41fR224):
```c#
    public sealed class HttpRequestOptions : IDictionary<string, object>
    {
        // Explicit interface implementation
        public bool TryGetValue<TValue>(HttpRequestOptionsKey<TValue> key, out TValue value);
        public void Set<TValue>(HttpRequestOptionsKey<TValue> key, TValue value);
    }
```

Reply:
Addressed in recent commits